### PR TITLE
[tf.data server] prevent restart of the dispatcher server w/o fault-tolerance, in the test bed

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR is a follow up of https://github.com/tensorflow/tensorflow/pull/43691 and prevents a restart of the dispatcher if it has been configured without fault-tolerance. Failing to do so will lead to a perpetual stream of `warn` messages on a dispatcher restart and unnecessarily extend the testing time without an immediate exit.

@aaudiber until the enhancements are implemented to handle such scenarios (as per our discussion), this will help us in handling the upcoming test cases.

### CODE REVIEW

File: tensorflow/python/data/experimental/kernel_tests/data_service_test_base.py [hunk 1]
```diff
@@ -115,7 +115,16 @@ def stop_dispatcher(self):
 
   # pylint: disable=protected-access
   def restart_dispatcher(self):
-    """Stops `dispatcher` and creates a new dispatcher with the same port."""
+    """Stops `dispatcher` and creates a new dispatcher with the same port.
+
+    When the dispatcher is restarted with `fault-tolerant_mode=False`,
```
Comment (aaudiber) [inline tensorflow/python/data/experimental/kernel_tests/data_service_test_base.py:120]: This scenario is too specific, can we make the docstring more general, e.g. "Restarting is only supported when the dispatcher is configured with `fault_tolerant_mode=True`"?

**Review by aaudiber — CHANGES_REQUESTED** [review]
Thanks @kvignesh1420

**Comment by kvignesh1420** [comment]
Made the change, @aaudiber.

**Review by aaudiber — APPROVED** [review]

**Comment by kvignesh1420** [comment]
@aaudiber, I had a quick question regarding the updates to the workers based on the task list.
Reference: https://github.com/tensorflow/tensorflow/blob/377434ff41473a33f7bedd1c08199d5af1fd33e9/tensorflow/core/kernels/data/experimental/data_service_dataset_op.cc#L326

How would we identify whether the task list changed or not if we don't poll for changes?
Please let me know.

**Pull request merged**